auth-tokens/src/main/java/com/palantir/tokens2/auth/Preconditions.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Copyright (C) 2007 The Guava Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.palantir.tokens2.auth;
+
+/*
+ * Derived from com.google.common.base.Preconditions
+ *
+ * Only a subset of the methods were copied.
+ * Small changes were made to respect this repository's code conventions.
+ * No change in behavior are expected from such changes.
+ */
+public final class Preconditions {
```

Comment:
I did not make it package private because this class was actually used in 2 different packages. One alternative could be to move this class to its own project and to use the api/implementation dependencies in the newish java-library gradle plugin. I've never used such plugin before though, so I might be missing something